app/controllers/CashAccountV2Controller.scala

```diff
@@ -100,14 +100,23 @@ class CashAccountV2Controller @Inject()(authenticate: IdentifierAction,
           appConfig.transactionsTimeoutFlag))
       }
 
-      case Right(cashTransactions) =>
-        if (cashTransactions.availableTransactions) {
-          Ok(
-            showAccountsView(form, CashAccountV2ViewModel(req.eori, account, cashTransactions))
-          )
-        } else {
-          Ok(noTransactionsWithBalance(CashAccountViewModel(req.eori, account)))
-        }
+      case Right(cashTransactions) => handleSuccessScenarios(cashTransactions, account)
+
+    }
+  }
+
+  private def handleSuccessScenarios(cashTransactions: CashTransactions,
+                                     account: CashAccount)(implicit req: IdentifierRequest[AnyContent]): Result = {
+
+    if (cashTransactions.maxTransactionsExceeded.contains(true)) {
+      Redirect(routes.CashAccountV2Controller.tooManyTransactions())
+    }
+    else if (cashTransactions.availableTransactions) {
+      Ok(
+        showAccountsView(form, CashAccountV2ViewModel(req.eori, account, cashTransactions))
```

Comment:
I am not sure this needs to be on three lines

Ok(showAccountsView(form, CashAccountV2ViewModel(req.eori, account, cashTransactions)))